### PR TITLE
ACM-16131: add back the clusterclaim kind and api version for the provisioner annotation

### PR DESCRIPTION
**Pull request by dtthuynh**
https://issues.redhat.com/browse/ACM-16131

### CODE REVIEW

**Comment by xiangjingli** [comment]
It looks good to me

@zhujian7  could you double review it 

@dtthuynh  Please verify the klusteraddonconfig CR is created along with the managed cluster creation. That could be handled in a e2e manual test after the PR is merged. It would be better to add a e2e test automation to cover this critical case

File: controllers/managedcluster/managedcluster-controller.go [hunk 1]
```diff
@@ -80,7 +80,7 @@ func (r *ManagedClusterReconciler) Reconcile(ctx context.Context, request ctrl.R
 	}
 
 	// annotation format is <name>.<namespace>
```
Comment (zhujian7) [inline controllers/managedcluster/managedcluster-controller.go:82]: nit: 
```suggestion
	// annotation format is <name>.<namespace>.<kind>.<apiversion>
```

**Comment by zhujian7** [comment]
LGTM

**Pull request merged**